src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Win32.cs

```diff
@@ -11,184 +10,167 @@ namespace System.IO
 {
     public partial class FileSystemWatcher
     {
+        /// <summary>Start monitoring the current directory.</summary>
         private void StartRaisingEvents()
         {
-            // If we're attached, don't do anything.
-            if (!IsHandleInvalid)
-            {
+            // If we're already running, don't do anything.
+            if (!IsHandleInvalid(_directoryHandle))
                 return;
-            }
 
             // Create handle to directory being monitored
-            _directoryHandle = NativeMethods.CreateFile(_directory,            // Directory name
-                                UnsafeNativeMethods.FILE_LIST_DIRECTORY,           // access (read-write) mode
-                                UnsafeNativeMethods.FILE_SHARE_READ |
-                                    UnsafeNativeMethods.FILE_SHARE_DELETE |
-                                    UnsafeNativeMethods.FILE_SHARE_WRITE,          // share mode
-                                null,                                              // security descriptor
-                                UnsafeNativeMethods.OPEN_EXISTING,                 // how to create
-                                UnsafeNativeMethods.FILE_FLAG_BACKUP_SEMANTICS |
-                                    UnsafeNativeMethods.FILE_FLAG_OVERLAPPED,      // file attributes
-                                new SafeFileHandle(IntPtr.Zero, false)             // file with attributes to copy
-                            );
-
-            if (IsHandleInvalid)
+            var defaultSecAttrs = default(Interop.mincore.SECURITY_ATTRIBUTES);
+            _directoryHandle = Interop.mincore.CreateFile(
+                lpFileName: _directory,
+                dwDesiredAccess: Interop.mincore.FileOperations.FILE_LIST_DIRECTORY,
+                dwShareMode: FileShare.Read | FileShare.Delete | FileShare.Write,
+                securityAttrs: ref defaultSecAttrs,
```

Comment:
No strong reason: the author of the original code apparently thought it was important to comment what these things were, so instead of comments I just used the parameter names (whenever possible I prefer using things the compiler will verify).